uberfire-security/uberfire-security-management/uberfire-security-management-backend/src/main/java/org/uberfire/ext/security/management/UberfireRoleManager.java

```diff
@@ -0,0 +1,106 @@
+import org.uberfire.ext.security.management.search.RuntimeSearchEngine;
+import org.uberfire.ext.security.management.util.SecurityManagementUtils;
+
+import javax.enterprise.context.Dependent;
+import javax.inject.Named;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+/**
+ * <p>The default role manager implementation for UF based applications.</p>
+ * <p>Roles are not coming from any external system, are just the ones registered in the application</p>
+ * <p>CRUD operations for roles are not allowed, only search is allowed.</p>
+ * 
+ * @since 0.8.0
+ */
+@Dependent
+@Named("uberfireRoleManager")
+public class UberfireRoleManager implements RoleManager, ContextualManager {
+
+    UserSystemManager userSystemManager;
+    RuntimeSearchEngine<Role> rolesSearchEngine;
+    
+    @Override
+    public void initialize(final UserSystemManager userSystemManager) throws Exception {
+        this.userSystemManager = userSystemManager;
+        rolesSearchEngine = new RolesRuntimeSearchEngine();
+    }
+    
+    @Override
+    public SearchResponse<Role> search(SearchRequest request) throws SecurityManagementException {
+        return rolesSearchEngine.search(getRegisteredRoles(), request);
+    }
+
+    @Override
+    public Role get(String identifier) throws SecurityManagementException {
+        throw new UnsupportedOperationException("Get operation is not available when using the UberfireRoleManager provider.");
+    }
+
+    @Override
+    public Role create(Role entity) throws SecurityManagementException {
+        throw new UnsupportedOperationException("Create operation is not available when using the UberfireRoleManager provider.");
+    }
+
+    @Override
+    public Role update(Role entity) throws SecurityManagementException {
+        throw new UnsupportedOperationException("Update operation is not available when using the UberfireRoleManager provider.");
+    }
+
+    @Override
+    public void delete(String... identifiers) throws SecurityManagementException {
+        throw new UnsupportedOperationException("Delete operation is not available when using the UberfireRoleManager provider.");
+    }
+
+    @Override
+    public RoleManagerSettings getSettings() {
+        return null;
+    }
+
+    @Override
+    public void destroy() throws Exception {
+
+    }
+
+    // Check that roles registered in the framework are created as realm groups as well, if not they cannot be used, as it couldn't be assigned.
+    protected Set<Role> getRegisteredRoles() {
+        final Set<Role> result = new LinkedHashSet<Role>();
+        final Set<Role> registeredRoles = SecurityManagementUtils.getRegisteredRoles();
+        if ( null != registeredRoles && !registeredRoles.isEmpty() ) {
+            GroupManager groupManager = userSystemManager.groups();
+            for (final Role registeredRole : registeredRoles) {
+                try {
+                    groupManager.get(registeredRole.getName());
```

Comment:
This method is used as well. Let me explain, when you want to assign roles to a user, you open the roles editor, which uses this role manager to obtain the system roles to show in the list for selection.

For the Tomcat and Keycloak, this method check that the roles "registered" (so in the RolesRegistry) do exist in the realm, but on the wildfly/eap case, which use the realms based on properties, registered roles could not be registered in the app-roles.properties, as this file does contain the roles, it contains only role assignments, so if a registered role is not in this file, it does not mean it doesn't exist, just that it has not been assigned yet. 

So for the wildfly/eap one this methods returns all registered roles, and for the other providers, the registered roles than exist in the realm (if the role does not exist, it cannot be assigned :)  )